guides/plugins/plugins/plugin-fundamentals/programmically-sending-emails.md

```diff
@@ -0,0 +1,119 @@
+# Programmically sending Emails
+
+In this guide you'll learn how to send Emails programmically using the `Mailservice` provided in Shopware v6.4.x.
+
+## Prerequisites
+
+In order to add the MailService for your plugin, you first need a plugin as base. Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md) and [how to add a custom controller](./add-custom-controller)
+
+## Registering the service
+
+Before we're working with the `MailSerivce` we need to register our controller to the [Dependency Injection](../../plugin-fundamentals/dependency-injection.md) container and inject the service `MailService` into our controller. We'll use a class called `ExampleController` here.
+
+{% code title="<plugin root>/src/Resources/config/services.xml" %}
+
+```markup
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <!-- Controller -->
+        <service id="Swag\Mail\Storefront\Controller\ExampleController" public="true">
```

Comment:
Is it neccessary to have the service public? I am just asking as I hadn't have to do this and now I am unsure whether my old 6.3 code will break in 6.4